pkg/controller/gcp/cache/claim.go

```diff
@@ -39,9 +39,25 @@ type CloudMemorystoreInstanceClaimController struct{}
 
 // SetupWithManager adds a controller that reconciles RedisCluster resource claims.
 func (c *CloudMemorystoreInstanceClaimController) SetupWithManager(mgr ctrl.Manager) error {
+	name := strings.ToLower(fmt.Sprintf("%s.%s.%s",
+		cachev1alpha1.RedisClusterKind,
+		v1alpha2.CloudMemorystoreInstanceKind,
+		v1alpha2.Group))
+
+	p := resource.NewPredicates(resource.AnyOf(
+		resource.HasManagedResourceReferenceKind(resource.ManagedKind(v1alpha2.CloudMemorystoreInstanceGroupVersionKind)),
+		resource.HasDirectClassReferenceKind(resource.NonPortableClassKind(v1alpha2.CloudMemorystoreInstanceClassGroupVersionKind)),
+		resource.HasIndirectClassReferenceKind(mgr.GetClient(), mgr.GetScheme(), resource.ClassKinds{
+			Portable:    cachev1alpha1.RedisClusterClassGroupVersionKind,
+			NonPortable: v1alpha2.CloudMemorystoreInstanceClassGroupVersionKind,
+		})))
+
 	r := resource.NewClaimReconciler(mgr,
 		resource.ClaimKind(cachev1alpha1.RedisClusterGroupVersionKind),
-		resource.ClassKinds{Portable: cachev1alpha1.RedisClusterClassGroupVersionKind, NonPortable: v1alpha2.CloudMemorystoreInstanceClassGroupVersionKind},
+		resource.ClassKinds{
```

Comment:
Since this is used here and in the predicates in each `SetupWithManager()`, how would you feel about declaring it as a var?

```go
classes := resource.ClassKinds{
  Portable:    cachev1alpha1.RedisClusterClassGroupVersionKind,
  NonPortable: v1alpha2.CloudMemorystoreInstanceClassGroupVersionKind,
}
```

Reply:
I went back and forth about this, but decided I slightly preferred explicitly writing it out in each case. I'm still pretty ambivalent and could be convinced that it would be better as a variable, but I'm too lazy to go update it in all three PRs so I'll leave it as is for now. ;)